zkvm/src/contract.rs

```diff
@@ -63,8 +83,7 @@ impl Contract {
             payload,
             anchor,
         };
-        let mut buf = Vec::with_capacity(contract.serialized_length());
-        contract.encode(&mut buf);
+        let mut buf = contract.encode_to_vec();
```

Comment:
```suggestion
        let buf = contract.encode_to_vec();
```

buffer is not mutable in this scope